tests/serverpod_test_server/test/generated_methods/from_json_test.dart

```diff
@@ -0,0 +1,378 @@
+      expect(
+        Types.fromJson({'anInt': 1}).anInt,
+        1,
+      );
+    });
+
+    test(
+        'when deserializing from JSON with an incorrect value type, then a TypeError is thrown',
+        () {
+      expect(
+        () => Types.fromJson({'anInt': '1'}).anInt,
+        throwsA(isA<TypeError>()),
+      );
+    });
+
+    test(
+        'when deserializing from JSON with a null value, then the integer field is set to null as expected',
+        () {
+      expect(
+        Types.fromJson({'anInt': null}).anInt,
+        null,
+      );
+    });
+  });
+
+  group('Given a class with a nullable double field, ', () {
+    test(
+        'when deserializing from JSON with a correct double value, then the double is deserialized correctly',
+        () {
+      expect(
+        Types.fromJson({'aDouble': 20.20}).aDouble,
+        20.20,
+      );
+    });
+
+    test(
+        'when deserializing from JSON with an incorrect value type, then a TypeError is thrown',
+        () {
+      expect(
+        () => Types.fromJson({'aDouble': '20.20'}).aDouble,
+        throwsA(isA<TypeError>()),
+      );
+    });
+
+    test(
+        'when deserializing from JSON with a null value, then the double field is set to null as expected',
+        () {
+      expect(
+        Types.fromJson({'aDouble': null}).aDouble,
+        null,
+      );
+    });
+  });
+
+  group('Given a class with a nullable boolean field, ', () {
+    test(
+        'when deserializing from JSON with a correct boolean value, then the boolean is deserialized correctly',
+        () {
+      expect(
+        Types.fromJson({'aBool': true}).aBool,
+        true,
+      );
+    });
+
+    test(
+        'when deserializing from JSON with an incorrect value type, then a TypeError is thrown',
+        () {
+      expect(
+        () => Types.fromJson({'aBool': 'true'}).aBool,
+        throwsA(isA<TypeError>()),
+      );
+    });
+
+    test(
+        'when deserializing from JSON with a null value, then the boolean field is set to null as expected',
+        () {
+      expect(
+        Types.fromJson({'aBool': null}).aBool,
+        null,
+      );
+    });
+  });
+
+  group('Given a class with a nullable string field, ', () {
+    test(
+        'when deserializing from JSON with a correct string value, then the string is deserialized correctly',
+        () {
+      expect(
+        Types.fromJson({'aString': 'test'}).aString,
+        'test',
+      );
+    });
+
+    test(
+        'when deserializing from JSON with an incorrect value type, then a TypeError is thrown',
+        () {
+      expect(
+        () => Types.fromJson({'aString': 111}).aString,
+        throwsA(isA<TypeError>()),
+      );
+    });
+
+    test(
+        'when deserializing from JSON with a null value, then the string field is set to null as expected',
+        () {
+      expect(
+        Types.fromJson({'aString': null}).aString,
+        null,
+      );
+    });
+  });
+
+  test(
```

Comment:
Suggestion: From here on down the "Group" pattern stops. Please apply groups for these tests as well.